turbopy/diagnostics.py

```diff
@@ -130,9 +130,86 @@ def _write_buffer(self):
             np.savetxt(f, self._buffer, delimiter=",")
 
 
+class NPYOutputUtility(OutputUtility):
+    """NumPy formatted binary file (.npy) diagnostic output helper class
+
+    Provides routines for writing data to a file in NumPy format. This
+    class can be used by Diagnostics subclassses to handle output to
+    .npy format.
+
+    Parameters
+    ----------
+    filename : str
+       File name for .npy data file.
+    diagnostic_size : (int, int)
+       Size of data set to be written to .npy file. First value is the
+       number of time points. Second value is number of spatial points.
+
+    Attributes
+    ----------
+    filename: str
+        File name for .npy data file.
+    buffer: :class:`numpy.ndarray`
+        Buffer for storing data before it is written to file.
+    buffer_index: int
+        Position in buffer.
+    """
+    
+    def __init__(self, filename, diagnostic_size, **kwargs):
+        self._filename = filename
+        self._buffer = np.zeros(diagnostic_size)
+        self._buffer_index = 0
+    
+    def diagnose(self, data):
+        """
+        Adds 'data' into npy output buffer.
+
+        Parameters
+        ----------
+        data : :class:`numpy.ndarray`
+            1D numpy array of values to be added to the buffer.
+        """
+        self._append(data)
+    
+    def finalize(self):
+        """Write the npy data to file.
+        """
+        self._write_buffer()
+    
+    def write_data(self):
+        """Write buffer to file"""
+        self._write_buffer()
+    
+    def append(self, data):
+        """Append data to the buffer.
+
+        .. deprecated::
+            `append` has been removed from the public API. Use `diagnose`
+            instead.
+        """
+        self._append(data)
```

Comment:
We don't need this method, since this is a new Class. Users can just call `diagnose` instead.

Please remove this function.